### ISSUE TITLE
Retrieve tree sorting by something other than default order

### ISSUE THREAD

**Issue by dmaksimov** [lazychaser/laravel-nestedset#85]
Would it be possible to have a way to pull the tree of nodes but sort it by another column.

Sometimes I need to pull all the nodes into an array but also order it alphabetically by the node's name.

I know I can use the toTree method, but having it all in one collection would be easier since I wouldn't have to recursively iterate through the children.

So the result would look something like the following.

- A parent
- - A child
- - B Child
- - C child
- B Parent
- - F child
- - - A child
- - - B Child
- - G child
- M Parent
- - A child
- - N child
- - O Child

So it would still pull the child nodes after the parent, but the child nodes under that parent would be in alphabetical order. The parents would also be in alphabetical order.

**Comment by lazychaser** [comment]
First of all, thanks for your support!

I don't see a query-based solution right now. You need to pull your nodes, convert them to tree, and then flatten it. Or reconsider your algorithm to use recursion in the first place.

**Comment by dmaksimov** [comment]
Yeah I figured it wouldn't be something that can be done with a query. It would be nice if the Collection class had a method to create a tree and then flatten it. 

I forked the repo and added some methods to do that so I can use it in my code. https://github.com/dmaksimov/laravel-nestedset/commit/ca678b31ce5e6784ef6070dcb496ce01684a1eec

If there's any way I can contribute, I'd love to. I'm still relatively new to Github and OOP in general so I'm still learning the ropes.

Let me know what you think.
